WordPress/gutenberg
# Bootstrap a minimal sync package

Related #52593 

## What?

The goal of this PR is to introduce a new package `@wordpress/sync` that will serve as a playground for us to iterate on the synchronization and live collaboration features. For the details about the architecture and the reasoning behind this package, please refer to this post https://make.wordpress.org/core/2023/07/13/real-time-collaboration-architecture/

It's most likely going to take a lot of time and iterations for us to arrive a decent API and experience for the sync engine and live collaboration, so it's important to have a starting point and a place to iterate without impacting the other work happening on the project, for this reason, all the features introduced by this work are hidden behind an experimental flag.

At the moment, the base architecture of the package is in place but it's not entirely hooked into all the necessary places:

 - The package relies on yjs to synchronize and merge changes.
 - Some entities of core-data have their corresponding "syncConfig" which allow their records to be synchronized.
 - The package allows offline indexeddb persistence (default yjs provider).
 - The package has a "non-working" WebRTC connector to set the API in place, It relies on the default yjs public signaling server but that server is not working properly at the moment. (So only cross tab collaboration is actually working)

## Testing Instructions

 1- Upgrade node
 2- Enable the feature flag
 3- you can open the same post in two different tabs and notice that changes propagate.
